README.md

```diff
@@ -299,58 +299,169 @@ not human-readable.
 Like the Base64, the binary format is only optimized for iteration.  Because of
 compression, records are inevitably of variable length, so random access is not
 possible.  Also, compression depends on iteration, as UUIDs get abbreviated
-relative to preceding UUIDs.
+relative to similar preceding UUIDs.
 
-A binary RON frame starts with magic bytes `RON2` and a big-endian uint32 frame length field, 8 bytes total.
+A binary RON frame starts with magic bytes `RON` and a frame *descriptor*
+specifying the version of the protocol and the length of the frame.  The rest
+of a frame is a sequence of *fields*.  Each field starts with a *descriptor*
+specifying the type of the field and its length.
 
-On the inside, a frame is a sequence of *fields*.  Each field starts with a
-*descriptor* byte.  A descriptor byte spends two most significant bits for a
-field type, next two bits for a sub-type and four bits for field byte length
-(excluding the descriptor byte, so starts with 0).  Length of 13, 14 or 15 means
-the descriptor byte is followed by the actual length as a big-endian uint8,
-uint16 or uint32, respectively.  Descriptor byte types and sub-types are as
-follows:
+### Descriptors
+
+A descriptor's first byte spends four most significant bits to describe the
+type of the field, other four bits describe its length.
+
+```
+   7    6    5    4    3    2    1    0
++----+----+----+----+----+----+----+----+
+| major   | minor   |len |field length/ |
+|    type |    type |flag| length length|
++----+----+----+----+----+----+----+----+
+  128  64   32   16    8    4    2    1
+```
+
+Frame descriptor type bits encode major/minor versions of the protocol (two
+m.s. bits for the major, other two for the minor, e.g. `10 00` for 2.0). 
```

Comment:
The proposed version must be 3.0 then.

Reply:
I did not break much of backward compatibility here :)

Reply:
You did break it. No matter how much.

But 2.* weren't in production maybe...